### PR TITLE
Implement some GameManager functions

### PR DESCRIPTION
**Pull request by Renzo904**
Implemented:
- GameManager's constructor
- GameManager::IncreaseSubrank
- GameManager::DecreaseSubrank

I also fixed the GameManager struct since the offsets were wrong(causing 358d9d9a07a1b43a43f52c1bdac77f048b9dd044 to break the pause menu), and added all the parameters from the ghidra repo

### CODE REVIEW

File: src/GameManager.cpp [hunk 1]
```diff
@@ -1,3 +1,48 @@
...
 
 DIFFABLE_STATIC(GameManager, g_GameManager);
+
+#pragma optimize("s", on)
+GameManager::GameManager()
+{
+
+    memset(this, 0, sizeof(GameManager));
+
+    (this->arcadeRegionTopLeftPos).x = 32.0;
```
Comment (roblabla) [inline src/GameManager.cpp:11]: I feel like we should put some #defines with those values somewhere, to make it easier to tweak later later. Something like `GAME_REGION_{TOP,LEFT,RIGHT,BUTTOM}`

File: src/GameManager.cpp [hunk 2]
```diff
@@ -1,3 +1,48 @@
...
+#pragma optimize("s", on)
+void GameManager::IncreaseSubrank(i32 amount)
+{
+    this->subRank = this->subRank + amount;
+    while (this->subRank >= 100)
+    {
+        this->rank++;
+        this->subRank = this->subRank - 100;
+    }
+    if ((this->rank > this->maxRank))
```
Comment (roblabla) [inline src/GameManager.cpp:27]:
```suggestion
    if (this->rank > this->maxRank)
```

File: src/GameManager.cpp [hunk 3]
```diff
@@ -1,3 +1,48 @@
...
+#pragma optimize("", on)
+
+#pragma optimize("s", on)
+void GameManager::IncreaseSubrank(i32 amount)
+{
+    this->subRank = this->subRank + amount;
+    while (this->subRank >= 100)
+    {
+        this->rank++;
+        this->subRank = this->subRank - 100;
```
Comment (roblabla) [inline src/GameManager.cpp:25]:
```suggestion
        this->subRank -= 100;
```

File: src/GameManager.cpp [hunk 4]
```diff
@@ -1,3 +1,48 @@
...
+#pragma optimize("", on)
+
+#pragma optimize("s", on)
+void GameManager::DecreaseSubrank(i32 amount)
+{
+    this->subRank = this->subRank - amount;
+    while (this->subRank < 0)
+    {
+        this->rank--;
+        this->subRank = this->subRank + 100;
```
Comment (roblabla) [inline src/GameManager.cpp:41]:
```suggestion
        this->subRank += 100;
```

**Review by roblabla — COMMENTED** [review]
Looking solid!

File: src/GameManager.cpp (hunk 1 above)
Comment (Renzo904) [reply src/GameManager.cpp:11]: Shouldn't be something more like GAME_REGION_{TOP, LEFT}, and then GAME_REGION_{WIDTH, HEIGHT}?

Comment (roblabla) [reply src/GameManager.cpp]: Ah yes, you're right.

**Pull request merged**